visual-espresso/visual/src/main/java/com/saucelabs/visual/VisualCheckOptions.java

```diff
@@ -1,81 +1,101 @@
+    private VisualCheckOptions(String testName, String suiteName, List<RegionIn> ignoreRegions, Boolean captureDom, DiffingOptionsIn diffingOptions) {
         this.testName = testName;
         this.suiteName = suiteName;
         this.ignoreRegions = ignoreRegions;
         this.captureDom = captureDom;
+        this.diffingOptions = diffingOptions;
     }
 
     public static final class Builder {
         private String testName;
         private String suiteName;
         private final List<RegionIn> ignoreRegions = new ArrayList<>();
         private Boolean captureDom;
+        private DiffingOptionsIn diffingOptions;
 
-        public Builder withTestName(String testName) {
+        public Builder testName(String testName) {
             this.testName = testName;
             return this;
         }
 
-        public Builder withSuiteName(String suiteName) {
+        public Builder suiteName(String suiteName) {
             this.suiteName = suiteName;
             return this;
         }
 
-        public Builder withIgnoreRegions(RegionIn... regions) {
-            this.ignoreRegions.addAll(Arrays.asList(regions));
+        public Builder ignoreRegions(Region... regions) {
+            for (Region region : regions) {
+                this.ignoreRegions.add(RegionInFactory.fromRegion(region));
+            }
             return this;
         }
 
         @SafeVarargs
-        public final Builder withIgnoreRegions(Matcher<View>... regions) {
-            List<RegionIn> ignoreRegions = new ArrayList<>();
-            for (Matcher<View> region : regions) {
-                GetRegionAction action = new GetRegionAction();
-                onView(region).perform(action);
-                ignoreRegions.add(action.getRegion());
+        public final Builder ignoreRegions(Matcher<View>... viewMatchers) {
+            List<RegionIn> result = new ArrayList<>();
+            for (Matcher<View> viewMatcher : viewMatchers) {
+                result.add(RegionInFactory.fromViewMatcher(viewMatcher));
+            }
+            this.ignoreRegions.addAll(result);
+            return this;
+        }
+
+        public Builder ignoreRegions(View... views) {
+            List<RegionIn> result = new ArrayList<>();
+            for (View view : views) {
+                result.add(RegionInFactory.fromView(view));
             }
-            this.ignoreRegions.addAll(ignoreRegions);
+            this.ignoreRegions.addAll(result);
             return this;
         }
 
-        public Builder withIgnoreRegions(View... regions) {
-            List<RegionIn> ignoreRegions = new ArrayList<>();
-            for (View view : regions) {
-                ignoreRegions.add(GetRegionAction.toRegion(view));
+        public Builder captureDom(boolean captureDom) {
+            this.captureDom = captureDom;
+            return this;
+        }
+
+        public Builder regions(SelectiveRegion... selectiveRegions) {
```

Comment:
With the backend diffing support not being available for these yet, maybe we should hold off on exposing this as a part of a public API?

Reply:
Yes that makes sense. I'll leave out capture dom as well

Reply:
actually captureDom is possible but we're not able to calculate the dom diff atm. I'll leave that one in the public API.